homeassistant/components/alexa/smart_home.py

```diff
@@ -383,8 +398,59 @@ def get_property(self, name):
             raise _UnsupportedProperty(name)
 
         unit = self.entity.attributes[CONF_UNIT_OF_MEASUREMENT]
+        temp = self.entity.state
+        if self.entity.domain == climate.DOMAIN:
+            temp = self.entity.attributes.get(
+                climate.ATTR_CURRENT_TEMPERATURE)
+        return {
+            'value': float(temp),
+            'scale': API_TEMP_UNITS[unit],
+        }
+
+
+class _AlexaThermostatController(_AlexaInterface):
+    def name(self):
+        return 'Alexa.ThermostatController'
+
+    def properties_supported(self):
+        properties = []
+        supported = self.entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE:
+            properties.append({'name': 'targetSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_LOW:
+            properties.append({'name': 'lowerSetpoint'})
+        if supported & climate.SUPPORT_TARGET_TEMPERATURE_HIGH:
+            properties.append({'name': 'upperSetpoint'})
+        if supported & climate.SUPPORT_OPERATION_MODE:
+            properties.append({'name': 'thermostatMode'})
+        return properties
+
+    def properties_retrievable(self):
+        return True
+
+    def get_property(self, name):
+        if name == 'thermostatMode':
+            mode = API_THERMOSTAT_MODES.get(self.entity.state)
```

Comment:
We should not use `entity.state`, instead, use `entity.attributes[climate.ATTR_OPERATION_MODE]`. Now you don't have to support `STATE_ON`/`STATE_OFF`.

Reply:
Ah, thanks! That leaves much fewer problematic platforms.